include/librealsense2/dds/dds-topic.cpp

```diff
@@ -0,0 +1,40 @@
+// License: Apache 2.0. See LICENSE file in root directory.
+// Copyright(c) 2022 Intel Corporation. All Rights Reserved.
+
+#include "dds-topic.h"
+#include "dds-participant.h"
+#include "dds-utilities.h"
+
+#include <fastdds/dds/domain/DomainParticipant.hpp>
+#include <fastdds/dds/topic/Topic.hpp>
+
+
+namespace librealsense {
+namespace dds {
+
+
+dds_topic::dds_topic( std::shared_ptr< dds_participant > const & participant,
+                      eprosima::fastdds::dds::TypeSupport const & topic_type,
+                      char const * topic_name )
+    : _participant( participant )
+    , _topic( nullptr )
+{
+    // REGISTER THE TYPE
+    DDS_API_CALL( topic_type.register_type( _participant->get() ) );
+
+    // CREATE THE TOPIC
+    _topic = DDS_API_CALL( _participant->get()->create_topic( topic_name,
+                                                              topic_type->getName(),
+                                                              eprosima::fastdds::dds::TOPIC_QOS_DEFAULT ) );
+}
+
+
+dds_topic::~dds_topic()
+{
+    if( _topic && _participant->is_valid() )
+        DDS_API_CALL_NO_THROW( _participant->get()->delete_topic( _topic ) );
```

Comment:
Creating the same topic twice is allowed, second time is ignored.
Need to check if can delete the same topic twice

Reply:
Actually, from looking at the code:
```
    if ((topics_.find(topic_name) != topics_.end()) ||
            (filtered_topics_.find(topic_name) != filtered_topics_.end()))
    {
        logError(PARTICIPANT, "Topic with name : " << topic_name << " already exists");
        return nullptr;
    }
```
Seems that creating a topic with the same name is an error.